cpp/oneapi/dal/algo/triangle_counting/backend/cpu/vertex_ranking_default_kernel.hpp

```diff
@@ -23,34 +23,250 @@
 #include "oneapi/dal/backend/interop/table_conversion.hpp"
 #include "oneapi/dal/detail/threading.hpp"
 #include "oneapi/dal/table/detail/table_builder.hpp"
+#include "oneapi/dal/algo/triangle_counting/backend/cpu/intersection_tc.hpp"
+#include "oneapi/dal/backend/primitives/intersection/intersection.hpp"
 
 namespace oneapi::dal::preview::triangle_counting::backend {
 
 template <typename Cpu>
-std::int64_t triangle_counting_global_scalar(const std::int32_t* vertex_neighbors,
-                                             const std::int64_t* edge_offsets,
-                                             const std::int32_t* degrees,
-                                             std::int64_t vertex_count,
-                                             std::int64_t edge_count);
+array<std::int64_t> triangle_counting_local(const dal::preview::detail::topology<std::int32_t>& t,
```

Comment:
would we have a copy of the code for 64 bit topology?

Reply:
The assumption is that kernel will differ from int32 algorithmically at least. If it will be the same -  the function can be extended with one template parameter. This is the level of backend which will not break backward compatibility.